contracts/contracts/SwapRouter.sol

```diff
@@ -0,0 +1,241 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity =0.7.6;
+pragma abicoder v2;
+
+import "@uniswap/v3-core/contracts/libraries/SafeCast.sol";
+import "@uniswap/v3-core/contracts/libraries/TickMath.sol";
+import "@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol";
+
+import "@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol";
+import "@uniswap/v3-periphery/contracts/base/PeripheryImmutableState.sol";
+import "@uniswap/v3-periphery/contracts/base/PeripheryValidation.sol";
+import "@uniswap/v3-periphery/contracts/base/PeripheryPaymentsWithFee.sol";
+import "@uniswap/v3-periphery/contracts/base/Multicall.sol";
+import "@uniswap/v3-periphery/contracts/base/SelfPermit.sol";
+import "@uniswap/v3-periphery/contracts/libraries/Path.sol";
+import "@uniswap/v3-periphery/contracts/libraries/PoolAddress.sol";
+import "@uniswap/v3-periphery/contracts/libraries/CallbackValidation.sol";
+import "@uniswap/v3-periphery/contracts/interfaces/external/IWETH9.sol";
+
+/// @title Uniswap V3 Swap Router
+/// @notice Router for stateless execution of swaps against Uniswap V3
+contract SwapRouter is
```

Comment:
Curious why you pulled this into its own file instead of using the version from `node_modules`? Did you make any changes from the original SwapRouter? 

Given that we should make some tweaks like going from `external` to `public` as mentioned above, I'm ok with this, but we should clearly document in the comments of this file what changes we made from the default SwapRouter

Reply:
Yeah so I looked there first, but I didn't see the contract there for some reason. This file should probably be moved elsewhere though until we have our own router. I didn't modify it all so far - will be sure to add comments when I do.

Reply:
would it be worth adding a dev note stating we forked the code from the link and we're using that

Reply:
Good call, yea in the comments we should:
- Link to the GitHub permalink (i.e. the URL containing the commit hash) this was copied from
- Document all changes from the original source (including that our prettier config was ran against it)

Reply:
should comments be somewhere at the top of the file? left a comment above the function but with these new changes, at the top might make more sense

Reply:
Yea if you could put a natspec comment at the top of the file that would be great! Check out `GrantRegistry.sol` and `MerkleGrantRoundPayout.sol` (you'll need to rebase against `main` for this one) for examples where we do this. The latter was also based on another Uniswap contract so might be a good comment template